lib/email_verifier/checker.rb

```diff
@@ -5,31 +5,29 @@ class EmailVerifier::Checker
 
   ##
   # Returns server object for given email address or throws exception
-  # Object returned isn't yet connected. It has internally a list of 
+  # Object returned isn't yet connected. It has internally a list of
   # real mail servers got from MX dns lookup
   def initialize(address)
     @email   = address
-    _, @domain  = address.split("@")
-    @servers = list_mxs @domain
+    @servers = list_mxs(address.split("@").last)
     raise EmailVerifier::NoMailServerException.new("No mail server for #{address}") if @servers.empty?
     @smtp    = nil
-
-    # this is because some mail servers won't give any info unless 
-    # a real user asks for it:
-    @user_email = EmailVerifier.config.verifier_email
-    _, @user_domain = @user_email.split "@"
   end
 
   def list_mxs(domain)
-    return [] unless domain
-    res = Dnsruby::DNS.new
+    return [] unless domain.present?
```

Comment:
We have to check if it's present some other way, since present? method is being added by Rails. We want to be independent of Rails while supporting it when it's present.